src/Core/Configurations/HostedRuntimeConfigProvider.cs

```diff
@@ -66,93 +56,25 @@ public RuntimeConfig GetConfig()
         {
             return _runtimeConfig;
         }
-
-        // While loading the config file, replace all the environment variables with their values.
-        if (ConfigLoader.TryLoadKnownConfig(out RuntimeConfig? config, replaceEnvVar: true))
-        {
-            _runtimeConfig = config;
-            TrySetupConfigFileWatcher();
-        }
-
-        if (_runtimeConfig is null)
+        else
         {
             throw new DataApiBuilderException(
                 message: "Runtime config isn't setup.",
                 statusCode: HttpStatusCode.ServiceUnavailable,
                 subStatusCode: DataApiBuilderException.SubStatusCodes.ErrorInInitialization);
         }
-
-        return _runtimeConfig;
     }
 
-    /// <summary>
-    /// Checks if we have already attempted to configure the file watcher, if not
-    /// instantiate the file watcher if we are in the correct scenario. If we
-    /// are not in the correct scenario, do not setup a file watcher but remember
-    /// that we have attempted to do so to avoid repeat checks in future calls.
-    /// Returns true if we instantiate a file watcher.
-    /// </summary>
-    private bool TrySetupConfigFileWatcher()
-    {
-        if (!IsLateConfigured && _runtimeConfig is not null && _runtimeConfig.IsDevelopmentMode())
-        {
-            try
-            {
-                FileSystemRuntimeConfigLoader loader = (FileSystemRuntimeConfigLoader)ConfigLoader;
-                _configFileWatcher = new(this, loader.GetConfigDirectoryName(), loader.GetConfigFileName());
-            }
-            catch (Exception ex)
-            {
-                // Need to remove the dependencies in startup on the RuntimeConfigProvider
-                // before we can have an ILogger here.
-                Console.WriteLine($"Attempt to configure config file watcher for hot reload failed due to: {ex.Message}.");
-            }
-
-            return _configFileWatcher is not null;
-        }
-
-        return false;
-    }
-
-    /// <summary>
-    /// Attempt to acquire runtime configuration metadata.
-    /// </summary>
-    /// <param name="runtimeConfig">Populated runtime configuration, if present.</param>
-    /// <returns>True when runtime config is provided, otherwise false.</returns>
+    // Not currently used in hosted scenario, but needed for interface
```

Comment:
comments need to explain why:
 `TryGetConfig([NotNullWhen(true)] out RuntimeConfig? runtimeConfig)`
 `public bool TryGetLoadedConfig([NotNullWhen(true)] out RuntimeConfig? runtimeConfig)`

 are not used in hosted scenario. It doesn't make sense to me that these two methods are promoted to the interface if they aren't used in implementating classes.